### ISSUE TITLE
Using HTTP::Cookie::URIParser causes error with 1.0.6

### ISSUE THREAD

**Issue by mtasaka** [sparklemotion/http-cookie#49]
With 1.0.6 `HTTP::Cookie::URIParser` module is added (mainly with https://github.com/sparklemotion/http-cookie/commit/052479536c0887e3d5bff1053515d3ac57545931 and https://github.com/sparklemotion/http-cookie/commit/8930674adf5b80c93e889bc2350d0a65f251fb5d ) 
but trying to use with 1.0.6 causes error like:

```
[tasaka1@localhost http-cookie-1.0.6]$ env RUBYLIB=$(pwd)/lib ruby -e "require 'http/cookie' ; p HTTP::Cookie::URIParser.escape_path('http://example.com/dir[]/file.html')"
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block (2 levels) in escape_path': uninitialized constant HTTP::Cookie::URIParser::CGI (NameError)

    path.sub(/\A[^?#]+/) { |p| p.gsub(/[:\[\]@]+/) { |r| CGI.escape(r) } }
                                                         ^^^
	from /home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `gsub'
	from /home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block in escape_path'
	from /home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `sub'
	from /home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `escape_path'
	from -e:1:in `<main>'
``` 

Also 1.0.6 testsuite fails:
```
[tasaka1@localhost http-cookie-1.0.6]$ ruby -Ilib:.:test -e 'Dir.glob("test/test_*.rb").each {|f| require f}' 
Loaded suite -e
Started
E
===============================================================================
Error: test_origin=(TestHTTPCookie): NameError: uninitialized constant HTTP::Cookie::URIParser::CGI
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block (2 levels) in escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `gsub'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block in escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `sub'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:33:in `rescue in parse'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:24:in `parse'
/home/mockuser/builddir/http-cookie-1.0.6/test/test_http_cookie.rb:952:in `test_origin='
     949:     }
     950: 
     951:     cookie = HTTP::Cookie.new('a', 'b')
  => 952:     cookie.origin = HTTP::Cookie::URIParser.parse('http://example.com/path[]/')
     953:     assert_equal '/path[]/', cookie.path
     954: 
     955:     cookie = HTTP::Cookie.new('a', 'b', :domain => '.example.com')
===============================================================================
E
===============================================================================
Error: test_valid_for_uri?(TestHTTPCookie): NameError: uninitialized constant HTTP::Cookie::URIParser::CGI
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block (2 levels) in escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `gsub'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `block in escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `sub'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:20:in `escape_path'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:33:in `rescue in parse'
/home/mockuser/builddir/http-cookie-1.0.6/lib/http/cookie/uri_parser.rb:24:in `parse'
/home/mockuser/builddir/http-cookie-1.0.6/test/test_http_cookie.rb:1037:in `test_valid_for_uri?'
     1034:         ]
     1035:       },
     1036:       HTTP::Cookie.parse('a4=b; domain=example.com; path=/dir2[]/',
  => 1037:         HTTP::Cookie::URIParser.parse('http://example.com/dir[]/file.html')).first => {
     1038:         true => [
     1039:           HTTP::Cookie::URIParser.parse('https://example.com/dir2[]/file.html'),
     1040:           HTTP::Cookie::URIParser.parse('http://example.com/dir2[]/file.html'),
===============================================================================
E
......
......
```

CI on github does not fail:
https://github.com/sparklemotion/http-cookie/actions/runs/9327065978/job/25676451401
but this is because testsuite uses `rake` , which loads `simplecov` , which loads `simplecov-html` , which loads `cgi` . So rake testsuite loads needed `cgi` indirectly.

**Comment by flavorjones** [comment]
@mtasaka Thank you for reporting this. Fix is at #50.

**Comment by flavorjones** [comment]
Fixed in https://github.com/sparklemotion/http-cookie/releases/tag/v1.0.7
